libvast/src/system/transformer.cpp

```diff
@@ -0,0 +1,83 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/system/transformer.hpp"
+
+#include "vast/detail/framed.hpp"
+#include "vast/detail/overload.hpp"
+#include "vast/error.hpp"
+#include "vast/logger.hpp"
+#include "vast/plugin.hpp"
+#include "vast/table_slice.hpp"
+
+#include <caf/attach_continuous_stream_stage.hpp>
+#include <caf/attach_stream_stage.hpp>
+#include <caf/settings.hpp>
+
+namespace vast::system {
+
+transformer_stream_stage_ptr attach_transform_stage(
+  transformer_actor::stateful_pointer<transformer_state> self) {
+  return caf::attach_continuous_stream_stage(
+    self,
+    [](caf::unit_t&) {
+      // nop
+    },
+    [self](caf::unit_t&, caf::downstream<table_slice>& out,
+           detail::framed<table_slice> x) {
+      if (x.header == detail::stream_control_header::eof) {
+        self->send_exit(self, caf::make_error(ec::end_of_input));
+        return;
+      }
+      auto transformed = self->state.transforms.apply(std::move(x.body));
+      if (!transformed) {
+        VAST_ERROR("discarding data: error in transformation step. {}",
+                   transformed.error());
+        return;
+      }
+      out.push(std::move(*transformed));
+    },
+    [=](caf::unit_t&, const caf::error&) {
+      // nop
+    });
+}
+
+transformer_actor::behavior_type
+transformer(transformer_actor::stateful_pointer<transformer_state> self,
+            std::string name, std::vector<transform>&& transforms) {
```

Comment:
You often pass names with underscore like `source_transformer` as the name, but we use dashes over underscores in actor names. Please change for consistent log output.